### PR TITLE
Configured edge router to accept connection from telemetry bench and …

### PR DESCRIPTION
**Pull request by pleimer**
In order to more realisticly test SAF, the setup needed to more closely reflect a real life setup in which data would come into an edge router on the client side before being routed over to the amq bus in SAF: tool -> qdr -> qdr -> sg.

I have added in a new router to satisfy this condition. The performance test automatically deploys and tracks the new router and points telemetry bench to send metrics through it.

### CODE REVIEW

File: tests/performance-test/deploy/performance-test-job-tb.yml.template [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
         - name: performance-test
           image: quay.io/redhat-service-assurance/telemetry-bench
           imagePullPolicy: Always
-          args: ["-hostprefix", "<<PREFIX>>", "-hosts", "<<HOSTS>>", "-plugins", "<<PLUGINS>>", "-instances", "1", "-send", "<<COUNT>>", "-interval", "<<INTERVAL>>", "-startmetricenable", "-verbose", "amqp://qdr-white.sa-telemetry.svc.cluster.local:5672/collectd/telemetry/"]
+          args: ["-hostprefix", "<<PREFIX>>", "-hosts", "<<HOSTS>>", "-plugins", "<<PLUGINS>>", "-instances", "1", "-send", "<<COUNT>>", "-interval", "<<INTERVAL>>", "-startmetricenable", "-verbose", "amqp://qdr-test.sa-telemetry.svc.cluster.local:5672/collectd/telemetry/"]
```
Comment (leifmadsen) [inline tests/performance-test/deploy/performance-test-job-tb.yml.template]: nice, this is a better naming scheme

File: tests/performance-test/deploy/qdrouterd.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      distribution: closest
+    - prefix: multicast
+      distribution: multicast
+    - prefix: unicast
+      distribution: closest
+    - prefix: exclusive
+      distribution: closest
+    - prefix: broadcast
+      distribution: multicast
+    - prefix: collectd
```
Comment (leifmadsen) [inline tests/performance-test/deploy/qdrouterd.yaml]: wouldn't this be collectd/telemetry?

File: tests/performance-test/deploy/qdrouterd.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      distribution: multicast
+    - prefix: collectd/notify
+      distribution: multicast
+  sslProfiles:
```
Comment (leifmadsen) [inline tests/performance-test/deploy/qdrouterd.yaml]: you don't need to specify a cert here, it should just accept what the interior router is using much like we do on the client side for openstack. There is just an `sslProfile` with a `name: default` and no certificates loaded. Then the connection is setup via SSL, but is implicitly authenticated.

Comment (leifmadsen) [reply tests/performance-test/deploy/qdrouterd.yaml]: I'm not exactly sure how all of that works in the Operator based deployment though, so it's possible that configuration for OpenShift isn't valid. Would need to check, but I suppose since you have access to the cert anyways, then why not?

In the future hopefully we'll get all that security stuff sorted out.

File: tests/performance-test/performance-test.sh [hunk 4]
```diff
@@ -35,17 +35,32 @@ ENDUSAGE
     exit 1
 }
 
+ELIPSE="."
```
Comment (leifmadsen) [inline tests/performance-test/performance-test.sh]: Can you add some docs on what this function in providing? I see that it changes one type of ellipses (note the spelling, should be 2 L's) to another, but it's not exactly clear at this point why this is necessary.

File: tests/performance-test/performance-test.sh [hunk 5]
```diff
@@ -58,6 +73,17 @@ create_service_monitors(){
     oc apply -f ./grafana/qdr-servicemonitor.yml
 }
 
+make_qdr_edge_router(){
```
Comment (leifmadsen) [inline tests/performance-test/performance-test.sh]: `make_qdr_edge_router_great_again()`

File: tests/performance-test/performance-test.sh [hunk 6]
```diff
@@ -141,28 +167,40 @@ while true; do
...
-                echo "Found new target endpoints. Creating performance test job"
+                echo "Found new target endpoints"
+                STAGE="ROUTER"
+            fi
+        ;;
+        "ROUTER")
+            make_qdr_edge_router
+            estab=$(oc get pod -l qdr_cr=qdr-test -o jsonpath='{.items[0].status.conditions[?(@.type=="ContainersReady")].status}') || true
+            if [ "${estab}" != "True" ]; then
+                printf "%s" "Waiting on qdr edge test pod creation"; elipses
```
Comment (leifmadsen) [inline tests/performance-test/performance-test.sh]: s/elipses/ellipses/g

File: tests/performance-test/performance-test.sh [hunk 7]
```diff
@@ -141,28 +167,40 @@ while true; do
...
+            else
+                echo "Qdr edge test pod established. Creating performance test job"
                 export COUNT HOSTS PLUGINS INTERVAL CONCURRENT
                 cd deploy
                 ./performance-test-tb.sh
                 STAGE="TEST"
             fi
         ;;
-        "TEST")
+        "TEST")
```
Comment (leifmadsen) [inline tests/performance-test/performance-test.sh]: trailing space

File: tests/performance-test/deploy/qdrouterd.yaml (hunk 2 above)
Comment (pleimer) [reply tests/performance-test/deploy/qdrouterd.yaml]: This I directly took from your config for the SAF router, so I figured I shouldn't change it?

File: tests/performance-test/deploy/qdrouterd.yaml (hunk 3 above)
Comment (pleimer) [reply tests/performance-test/deploy/qdrouterd.yaml]: You're right. I used this when I was trying to debug something before I understood all that the operator was doing

File: tests/performance-test/deploy/qdrouterd.yaml (hunk 2 above)
Comment (leifmadsen) [reply tests/performance-test/deploy/qdrouterd.yaml]: Oh really? Hrmmm... something seems off here then. I think Chris updated this to `collectd/telemetry` and `collectd/notify` for the metrics vs events addresses.

For testing purposes, whatever works I suppose, but I would have expected this to be `telemetry` not just `collectd`. I could be misunderstanding though.

File: tests/performance-test/deploy/qdrouterd.yaml (hunk 3 above)
Comment (pleimer) [reply tests/performance-test/deploy/qdrouterd.yaml]: It's asymmetrical authentication so there isn't need for that here, just remnant code I accidently left in

File: tests/performance-test/performance-test.sh (hunk 4 above)
Comment (pleimer) [reply tests/performance-test/performance-test.sh]: For sure. Really just to make it clear that the script is not frozen when it is waiting on pods to deploy. If an image must be pulled from an online repository, it can take a while

File: tests/performance-test/performance-test.sh (hunk 6 above)
Comment (pleimer) [reply tests/performance-test/performance-test.sh]: oops

File: tests/performance-test/deploy/qdrouterd.yaml (hunk 2 above)
Comment (pleimer) [reply tests/performance-test/deploy/qdrouterd.yaml]: It works either way so I will go ahead and change it so we don't get unintended messages skewing our data later on

**Comment by csibbitt** [comment]
[Make more realistic test tool setup (tool -> qdr -> qdr -> sg)](https://trello.com/c/XMZ5et62/682-make-more-realistic-test-tool-setup-tool-qdr-qdr-sg)
